backend/LoadCell.py

```diff
@@ -0,0 +1,104 @@
+from enum import Enum
+import matplotlib.pyplot as plt
+import json
+import numpy as np
+
+class CalLoadCell():
+    class State(Enum):
+        NORMAL = 0
+        STORE = 1
+        WAIT = 2
+
+    def __init__(self, name="DefaultLoadCell") -> None:
+        self.name = name
+        self.load_cell = 0
+        self.tare = 0
+        self.slope = 1
+        self.vol_offset = 0
+        self.cal = False
+        self.cal_points = {}
+        self.mass = 0
+        self.volts = 0
+        self.txtval = 0
+        self.current_state = self.State.NORMAL
+
+    def reset(self):
+        self.current_state = self.State.NORMAL
+
+    def _collect_point(self, voltage, mass):
+        self.current_state: self.State.NORMAL
+        if not self.cal_points:
+            self.cal_points = {0: voltage}
+            self.vol_offset = voltage
+        else:
+            self.cal_points[mass] = voltage
+
+    def _discard_cal(self, num=1):
+        for i in range(num):
+            self.cal_points.popitem()
+
+    def _display(self):
+        masses = list(self.cal_points.keys())
+        voltages = list(self.cal_points.values())
+        plt.scatter(voltages, masses, label='Calibration Points')
+        plt.xlabel('Voltage')
+        plt.ylabel('Mass')
+        plt.title('Calibration Points and Linear Regression')
+        if self.cal:
+            x_values = [min(voltages), max(voltages)]
+            y_values = [self.calc_weight(x) for x in x_values]
+            plt.plot(x_values, y_values, color='red', label='Linear Regression')
+        plt.legend()
+        plt.show()
+
+    def _store(self):
+        with open("val.txt", 'w') as r:
+            self.txtval = int(r.read()) + 1
+            r.write(self.txtval)
+        jsonValues = {
+            "LoadCell": "f{self.name}+{self.txtval}",
+            "Tare": self.tare,
+            "Slope": self.slope,
+            "Offset": self.vol_offset,}
+        json_data = json.dumps(jsonValues, indent=1)
+        with open("data.json", "w") as writer:
+            writer.write(json_data)
+
+    def _load(self):           
+        with open("data.json" 'r') as reader:
+            data = json.load(reader)
+            self.name = data["LoadCell"]
+            self.tare = data["tare"]
+            self.slope = data["Slope"]
+            self.vol_offset = data["Offset"]
+
+    def cal_fin(self):
```

Comment:
NIT: this is more personal preference so I'll put is as NIT but for the most part I try to do full words for the names of variables and functions, so this would be `calibation_finish` or `finalize`, and then in `__init__()` I would make `vol_offset` `voltage_offset`

Reply:
+1

Reply:
I made the changes on the functions so that they have either full names or use better names to indicate what they are doing